repos/c2sm/packages/icon4py/package.py

```diff
@@ -0,0 +1,115 @@
+import json
+import os
+import pathlib
+
+import llnl
+from llnl.util import tty
+from spack import *
+
+
+class Icon4py(Package):
+    extends("python")
+    depends_on("python@3.11:")
```

Comment:
Did I misinterpret the constraint in https://github.com/C2SM/spack-c2sm/pull/1059/files#r2451548916 or does this conflict with it?

Reply:
Also in the `Ghex` package, `python@3.7` is only a build dependency, so hopefully does not conflict with `Icon4py`.